packages/graph/hash_graph/tests/integration/postgres/mod.rs

```
@@ -0,0 +1,150 @@
+
+            let mut connection = PgConnection::connect(&connection_info.url())
+                .await
+                .expect("could not connect to database");
+
+            let account_id = create_account_id(&mut connection).await;
+
+            (postgres, connection, account_id)
+        });
+
+        Self {
+            postgres,
+            connection: Some(connection),
+            created_base_uris: Vec::new(),
+            account_id,
+            rt,
+        }
+    }
+
+    pub fn create_data_type(
+        &mut self,
+        data_type: DataType,
+    ) -> Result<Qualified<DataType>, InsertionError> {
+        self.rt.block_on(async {
+            let data_type = self
+                .postgres
+                .create_data_type(data_type, self.account_id)
+                .await?;
+            self.created_base_uris
+                .push(data_type.inner().id().base_uri().clone());
+            Ok(data_type)
+        })
+    }
+
+    pub fn get_data_type(
+        &mut self,
+        version_id: VersionId,
+    ) -> Result<Qualified<DataType>, QueryError> {
+        self.rt
+            .block_on(async { self.postgres.get_data_type(version_id).await })
+    }
+
+    pub fn update_data_type(
+        &mut self,
+        data_type: DataType,
+    ) -> Result<Qualified<DataType>, UpdateError> {
+        self.rt.block_on(async {
+            self.postgres
+                .update_data_type(data_type, self.account_id)
+                .await
+        })
+    }
+}
+
+impl Drop for Database {
+    fn drop(&mut self) {
+        let mut connection = self.connection.take().unwrap();
+        let account_id = self.account_id;
+        let created_base_uris = self.created_base_uris.clone();
+
+        self.rt.block_on(async move {
+            for base_uri in created_base_uris {
+                remove_by_base_uri(&mut connection, &base_uri).await;
+            }
+            remove_account_id(&mut connection, account_id).await;
```

Do we want to panic here or do we want to ignore errors?